Hide UI elements when Flipper is in read-only mode

radar opened the pull request:
We have a particular business-critical application that uses Flipper flags, and we would like to provide a completely read-only UI with Flipper. I know of a way to do this, which is to set `Flipper::UI`'s configuration to be read only. However, this same application is a bit a "nexus" for some smaller apps, and therefore we have multiple instances of `Flipper::UI` at play here. The different instances each control the flipper flags for the different applications. 

I was able to see that there's a `ReadOnly` adapter however:

```
readonly_app = Flipper::UI.app(Flipper.new(Flipper::Adapters::ReadOnly.new(connection)))
```

But this doesn't hide all the UI buttons in Flipper UI. Here's what I see in that app by default:

![image](https://github.com/flippercloud/flipper/assets/2687/1c19af02-939e-489e-9b02-448f2c1b253f)

Having all the buttons and forms present makes our security people uncomfortable.

---

So what I've tried to do in this PR is to cut out all of the UI that performs a write action, which then cuts the page down to this:

![image](https://github.com/flippercloud/flipper/assets/2687/02b3a688-6c44-4984-ad06-155fa32cbfd5)

And the homepage to this:

![image](https://github.com/flippercloud/flipper/assets/2687/c8740795-af49-4163-93f3-9b133e10c2fd)

I've also updated the "nooooope" error screen to show that the `ReadOnly` adapter might be in use:

![image](https://github.com/flippercloud/flipper/assets/2687/dd0c8973-0166-4f15-a761-42418c1b1418)

radar commented:
Thanks for making the Flipper code so easy to read and reason about. It was a pleasure to contribute :)

jnunemaker commented on lib/flipper/ui/actions/actors_gate.rb:
```diff
@@ -23,7 +23,7 @@ def get
 
         def post
-          read_only if Flipper::UI.configuration.read_only
+          read_only if read_only?
```
Should this be render_read_only also?

jnunemaker commented on lib/flipper/ui/views/features.erb:
```diff
@@ -3,15 +3,16 @@
       <h4>But I've got a blank space baby...</h4>
       <p>And I'll flip your features.</p>
-      <%- if !Flipper::UI.configuration.read_only && Flipper::UI.configuration.feature_creation_enabled -%>
+      <%= read_only? %>
```
Did you mean to leave this?

jnunemaker commented on spec/flipper/ui/actions/add_feature_spec.rb:
```diff
@@ -39,4 +39,24 @@
     end
   end
+
+  describe 'GET /features/new when an adpter is set to readonly' do
```
```suggestion
  describe 'GET /features/new when an adapter is set to readonly' do
```

jnunemaker reviewed (commented):
Looks great. I left a couple questions. If you have time to answer/address go for it. If you don't, just let me know and I'll get this over the finish line.

radar commented:
1. Removed the leftover view code.
2. Fixed up the add feature spec which wasn't actually using the `read_only?` query method.
3. Added a regression test for ActorsGate, and then fixed the code.
